src/clients/ntp-client_proposal.rb

```diff
@@ -514,16 +532,34 @@ def dhcp_ntp_items
     #
     # @return [Array<Yast::Term>] ntp address table Item
     def fallback_ntp_items
-      dhcp_items = dhcp_ntp_items
-
-      log.info("Nothing found in /etc/chrony.conf")
-      if dhcp_items.empty?
+      return @cached_fallback_ntp_items if @cached_fallback_ntp_items
+      @cached_fallback_ntp_items = dhcp_ntp_items
+      if !@cached_fallback_ntp_items.empty?
+        log.info("Proposing NTP server list provided by DHCP")
+      else
         log.info("Proposing current timezone-based NTP server list")
-        return timezone_ntp_items
+        @cached_fallback_ntp_items = timezone_ntp_items
       end
+      @cached_fallback_ntp_items
+    end
 
-      log.info("Proposing NTP server list provided by DHCP")
-      dhcp_items
+    # Checking if the user can select one ntp server from the list
+    # of proposed servers.
+    # It does not make sense if there are more than one ntp server has been
+    # defined.
+    #
+    # @return [Boolean] true if the user should select a server
+    def select_ntp_server
+      ret = NtpClient.GetUsedNtpServers.nil? || NtpClient.GetUsedNtpServers.empty?
+      # It could be that the user has defined an own ntp server in the ntp-client
+      # module which is not defined in the combo box. In that case we do not offer
+      # a selection. The user should go back to ntp-client to change it.
+      if NtpClient.GetUsedNtpServers.size == 1
+        ret = fallback_ntp_items.any? do |item|
+          item.params[1] == NtpClient.GetUsedNtpServers.first
+        end
+      end
+      ret
```

Comment:
well, this part of code is already big and not tested at all. And this change again increase it. I think we should consider moving this part of code somewhere where it can be tested and also some refactoring is welcome.

Reply:
ah, it is tested, so it is a bit better. Still I think it would be nice to not increase this file and instead e.g. create own dialog class that is used from this client.

Reply:
Well, espl. for that part I have added testcases in order to be sure.

Reply:
Well this is an L3. I do not want a refactoring. 
To be honest this file has to be refactored completely. This already begins with the complete wrong file name and ends in the code. But first of all we have to decide if the connection of time settings and ntp-client configuration makes sense at all. At first we should decide how the UI should be before starting a refactoring.